include/pelz_json_parser.h

```diff
@@ -78,7 +78,7 @@ int message_encoder(RequestType request_type, charbuf key_id, charbuf data, char
  * @return 0 on success, 1 on error
  *
  */
-int encrypt_parser(cJSON * json, charbuf * key_id, charbuf * data);
+int encrypt_parser(cJSON * json, charbuf * key_id, charbuf * data, charbuf * request_sig, charbuf * requestor_cert);
```

Comment:
If the certificate and signature over the request were both validated in request_decoder(), the additional parameters (request_sig and requestor_cert) would not need to be passed to the lower level parsers (encrypt_parser and decrypt_parser).  It does not have to be done this way, but I think it might be cleaner to validate a signed request in request_decoder() and then, if validation is successful (or the request is unsigned), the encrypt_parser() or decrypt_parser() functionality could then be called without modification.